chef/cookbooks/neutron/recipes/common_agent.rb

```diff
@@ -202,10 +203,37 @@
       command "ovs-vsctl add-br #{name}; ip link set #{name} up"
       not_if "ovs-vsctl list-br |grep -q #{name}"
     end
-    execute "Neutron: add #{bound_if} to #{name}" do
-      command "ovs-vsctl del-port #{name} #{bound_if} ; ovs-vsctl add-port #{name} #{bound_if}"
-      not_if "ovs-dpctl show system@#{name} | grep -q #{bound_if}"
+
+    # If:
+    #   * this network is on the same physical interface as the admin network
+    #   * we're running in HA mode
+    #   * corosync is configured to use multicast networking
+    # We need to stop openais/corosync before moving the network interface
+    # into the openvswitch. Otherwise we'll break corosync's multicast
+    # communication and cause fencing.
+    # For details: https://bugzilla.suse.com/show_bug.cgi?id=915176
+    if (node[:crowbar_wall][:network][:nets]["admin"].last == bound_if &&
+        neutron_l3_ha && node[:corosync][:transport] == "udp" )
+
+      # Put the whole cluster into maintenance mode before shutting down
+      # corosync. As this operation affects all cluster nodes only do it on
+      # the founder.
+      if (CrowbarPacemakerHelper.is_cluster_founder?(node))
+        execute "enable maintenance mode" do
+          command "crm --wait configure property maintenance-mode=true"
```

Comment:
Are you sure this puts the whole cluster in maintenance mode, and not just this node?